### PR TITLE
 ref(metrics): Cleans up metrics API [INGEST-875]

### PR DESCRIPTION
**Pull request by ahmedetefy**
Hacks out the `MockingDataSource`, and
 moves the `SnubaDataSource` into its own
 module

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_metrics.py [hunk 1]
```diff
@@ -54,222 +50,29 @@ def test_feature_flag(self):
             assert response.status_code == 404
 
 
-class OrganizationMetricsTest(APITestCase):
```
Comment (jjbayer) [inline tests/sentry/api/endpoints/test_organization_metrics.py:57]: Should we maybe keep these tests, but with some mock data in Snuba?

File: src/sentry/snuba/metrics/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from .datasource import *  # NOQA
+from .helpers import *  # NOQA
```
Comment (jjbayer) [inline src/sentry/snuba/metrics/__init__.py:2]: Thanks for this split! We might even split `.datasource` into `.data` and `.meta`, to make it obvious which functions are involved in metadata requests and which are used for totals & time series.

File: src/sentry/snuba/metrics/helpers.py [hunk 3]
```diff
@@ -0,0 +1,694 @@
...
+    _BASE_TAGS,
+    transaction=["/foo/:orgId/", "/bar/:orgId/"],
+)
+
+_MEASUREMENT_TAGS = dict(
+    _TRANSACTION_TAGS,
+    measurement_rating=["good", "meh", "poor"],
+)
+
+METRICS = {
```
Comment (jjbayer) [inline src/sentry/snuba/metrics/helpers.py:410]: I would actually like to get rid of this entire dict, but let's do that in a follow up PR. I will create a Jira issue for that.

Comment (jjbayer) [reply src/sentry/snuba/metrics/helpers.py:410]: Oh wait, is this now only used in `_get_metric`? In that case I would remove it right away. We don't need to return a 400 response if the metric does not exist in snuba, an empty result is fine.

File: tests/sentry/api/endpoints/test_organization_metrics.py (hunk 1 above)
Comment (ahmedetefy) [reply tests/sentry/api/endpoints/test_organization_metrics.py:57]: They pretty much do the exact same thing as the integration ones, didn't really see the need to duplicating them, and the ones that did something different I kept

File: src/sentry/snuba/metrics/__init__.py (hunk 2 above)
Comment (ahmedetefy) [reply src/sentry/snuba/metrics/__init__.py:2]: Np! I don't know if that's necessary `.data` module would just have one function that is `get_series`. I do not see the need to do this split, atleast not now

File: tests/sentry/api/endpoints/test_organization_metrics.py (hunk 1 above)
Comment (jjbayer) [reply tests/sentry/api/endpoints/test_organization_metrics.py:57]: OK if the same functionality is still covered, then that makes sense.

File: src/sentry/snuba/metrics/helpers.py (hunk 3 above)
Comment (ahmedetefy) [reply src/sentry/snuba/metrics/helpers.py:410]: Yeah it seems that's the only place its used for, I agree with your suggestions.. so how would you edit `_get_metric` ?

Comment (jjbayer) [reply src/sentry/snuba/metrics/helpers.py:410]: I would remove it entirely and use the function name (`op`) to map to the correct metrics type here instead:
https://github.com/getsentry/sentry/blob/ddb6ef15e2036ac4c9953a2266224b2fe983d992/src/sentry/snuba/metrics/helpers.py#L532-L534

If you think this is out of scope for the PR, I'm happy to do it in a follow-up.

Comment (jjbayer) [reply src/sentry/snuba/metrics/helpers.py:410]: I would remove it entirely and use the function name (`op`) to map to the correct metrics type here instead:
https://github.com/getsentry/sentry/blob/ddb6ef15e2036ac4c9953a2266224b2fe983d992/src/sentry/snuba/metrics/helpers.py#L532-L534

If you think this is out of scope for the PR, I'm happy to do it in a follow-up.

Comment (ahmedetefy) [reply src/sentry/snuba/metrics/helpers.py:410]: Will do a follow up PR like we discussed on slack

**Review by jjbayer — APPROVED** [review]

**Pull request merged**